src/blenderbim/blenderbim/bim/import_ifc.py

```diff
@@ -684,6 +686,62 @@ def create_structural_point_connections(self):
             obj.matrix_world = self.apply_blender_offset_to_matrix_world(obj, placement_matrix)
             self.link_element(product, obj)
 
+    def create_pointclouds(self, products):
```

Comment:
Can I be a pain and recommend a different approach?

```python
def create_point_clouds(self, products):
    for product in products:
        representation = self.get_pointcloud_representation(product)
        if representation:
            self.create_point_cloud(product, representation)

def create_point_cloud(self, product, representation):
    # since we already have the representation, we no longer need to duplicate the code for checking representation type https://github.com/IfcOpenShell/IfcOpenShell/pull/1804/files#diff-84a51bb6e3dd5dcf1060381cfa319cd64e6f4b819e5208387d0e74d73dcf77a5R696-R699
    # also, we just loop through items directly, no need to create another list for items so this variable disappears https://github.com/IfcOpenShell/IfcOpenShell/pull/1804/files#diff-84a51bb6e3dd5dcf1060381cfa319cd64e6f4b819e5208387d0e74d73dcf77a5R694
    # also https://github.com/IfcOpenShell/IfcOpenShell/pull/1804/files#diff-84a51bb6e3dd5dcf1060381cfa319cd64e6f4b819e5208387d0e74d73dcf77a5R713-R717 can be deleted and just use mesh_name = f"{representation.ContextOfItems.id()}/{representation.id()}"
    # If you self.file.traverse each representation item, this code can be deleted too https://github.com/IfcOpenShell/IfcOpenShell/pull/1804/files#diff-84a51bb6e3dd5dcf1060381cfa319cd64e6f4b819e5208387d0e74d73dcf77a5R719-R720
    # Rest of the code is the same as what you've written
```